src/createServer.js

```diff
@@ -1,8 +1,51 @@
 'use strict';
 
+const http = require('http');
+const fs = require('fs');
+const path = require('path');
+
+const FILE_PATH = '/file';
+const PUBLIC_PATH = '/public';
+
 function createServer() {
-  /* Write your code here */
-  // Return instance of http.Server class
+  const server = http.createServer((req, res) => {
+    const { pathname } = new URL(req.url, `http://${req.headers.host}`);
+
+    res.setHeader('Content-Type', 'text/plain');
+
+    if (!pathname.includes(FILE_PATH)) {
```

Comment:
using `startsWith` method is a bit safer way to check it

Reply:
agry!

Reply:
https://github.com/mate-academy/node_static-files-server/pull/172#issue-2103100003
Active question

Reply:
when it comes to your question: only files from the public folder should be returned, so we should make sure that we do not return anything that is located outside.

Not using `path.join()` to resolve the path should be enough to prevent this behavior.
```javascript
fs.readFile(`./public/../someFile.js, ...)
```

Reply:
done